extensions/vscode/src/commands/debug-dev-server.ts

```diff
@@ -0,0 +1,128 @@
+import { Uri, commands, debug, extensions, window } from "vscode";
+import {
+  isDartFrogCLIInstalled,
+  quickPickApplication,
+  suggestInstallingDartFrogCLI,
+} from "../utils";
+import { DartFrogApplication, DartFrogDaemon } from "../daemon";
+
+export const debugDevServer = async (): Promise<void> => {
+  if (!isDartFrogCLIInstalled()) {
+    await suggestInstallingDartFrogCLI(
+      "Running this command requires Dart Frog CLI to be installed."
+    );
+  }
+
+  const dartExtension = extensions.getExtension("Dart-Code.dart-code");
+  if (!dartExtension) {
+    const selection = await window.showErrorMessage(
+      "Running this command requires the Dart extension.",
+      "Install Dart extension",
+      "Cancel"
+    );
+    switch (selection) {
+      case "Install Dart extension":
+        const dartCodeMarketplaceUri =
+          "https://marketplace.visualstudio.com/items?itemName=Dart-Code.dart-code";
+        commands.executeCommand(
+          "vscode.open",
+          Uri.parse(dartCodeMarketplaceUri)
+        );
+        return;
+      case "Cancel":
+      default:
+        return;
+    }
+    /* c8 ignore start */
+  }
```

Comment:
For some reason this closing } line is not covered by the coverage tooling. I have my suspicion that is related to ts transpilation to js and reporting back. I personally, don't think this is important for this Pull Request to land and I'm planning to create an issue to track this problem.

Reply:
Tracking issue https://github.com/VeryGoodOpenSource/dart_frog/issues/961